backend/siarnaq/api/compete/views.py

```diff
@@ -236,13 +236,34 @@ def get_queryset(self):
         )
         return queryset
 
+    def get_serializer_context(self):
+        context = super().get_serializer_context()
+        if self.action != "tournament":
+            return context
+        # check if the external id is valid
+        external_id_private = self.request.query_params.get("external_id_private")
+        tournaments = None
+        if external_id_private is None:
+            return context
+        tournaments = Tournament.objects.visible_to_user(is_staff=True)
+        tournaments = tournaments.filter(external_id_private=external_id_private)
+        if tournaments.count() == 1:
+            # if the external id is valid, allow lookup of all tournament info
+            context.update({"user_is_staff": True})
+        return context
+
     @extend_schema(
         parameters=[
             OpenApiParameter(
                 name="tournament_id",
                 type=str,
                 description="A tournament to filter for.",
             ),
+            OpenApiParameter(
+                name="external_id_private",
+                type=str,
+                description="A private id to filter for.",
```

Comment:
Maybe this should be described as having to do with Challonge? Or is it like this in case we switch away from Challonge?

Reply:
Yeah, it's meant to be generic